yt_dlp/extractor/nebula.py

```diff
@@ -51,7 +50,7 @@ def _perform_nebula_auth(self):
 
         return response['key']
 
-    def _retrieve_nebula_api_token(self):
+    def _retrieve_nebula_api_token(self, username=None, password=None):
```

Comment:
```suggestion
    def _retrieve_nebula_api_token(self, username, password):
```

Reply:
Even though that signature is currently never called, I think it's valid for this function to be called without explicit credentials. In that case, it only attempts authentication via cookies and only if those cannot be found, `raise_login_required()` will be raised.

Reply:
There is really no point in having code that is never used, but I don't care enough to argue about this